typespec-tests/src/test/java/com/type/union/UnionsClientTest.java

```diff
@@ -3,56 +3,64 @@
 
 package com.type.union;
 
-import com.azure.core.http.rest.Response;
 import com.azure.core.util.BinaryData;
+import com.type.union.models.Model1;
+import com.type.union.models.Model2;
+import com.type.union.models.ModelWithNamedUnionProperty;
+import com.type.union.models.ModelWithNamedUnionPropertyInResponse;
+import com.type.union.models.ModelWithSimpleUnionProperty;
+import com.type.union.models.ModelWithSimpleUnionPropertyInResponse;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+import java.util.Arrays;
+import java.util.List;
+
 class UnionsClientTest {
 
     private final UnionClient client = new UnionClientBuilder().buildClient();
 
     @Test
     void sendInt() {
-        client.sendIntWithResponse(BinaryData.fromString("{ \"simpleUnion\": 1 }"), null);
+        client.sendInt(new ModelWithSimpleUnionProperty(BinaryData.fromObject(1)));
```

Comment:
Here again, an age-old argument of `BinaryData` vs. `Object`.

In setter, `Object` is simpler, one can write `client.sendInt(new ModelWithSimpleUnionProperty(1));` here. No need for `BinaryData.from##`.

But in response, `BinaryData` give user the ability to deserialize to the exact class.